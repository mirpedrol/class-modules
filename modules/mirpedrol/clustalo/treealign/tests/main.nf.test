nextflow_process {

    name "Test Process CLUSTALO_TREEALIGN"
    script "../main.nf"
    process "CLUSTALO_TREEALIGN"

    tag "modules"
    tag "modules_mirpedrol"
    tag "clustalo"
    tag "clustalo/treealign"
    tag "clustalo/guidetree"

    test("sarscov2 - contigs-fasta - guide_tree") {

        setup {

            run("CLUSTALO_GUIDETREE") {
                script "../../guidetree/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ], // meta map
                                 file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                               ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                             file(params.test_data['sarscov2']['illumina']['contigs_fasta'], checkIfExists: true)
                           ]
                input[1] = CLUSTALO_GUIDETREE.out.tree.collect{ meta, tree -> tree }.map{ tree -> [[ id: 'test_summary'], tree]}
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.alignment).match("with_guide_tree_alignment")},
                { assert snapshot(process.out.versions).match("with_guide_tree_versions") }
            )
        }
    }
}