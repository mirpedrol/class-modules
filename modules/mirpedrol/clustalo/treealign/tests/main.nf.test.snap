{
    "with_guide_tree_alignment": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.aln.gz:md5,74bb9a2820a91cf68db94dbd46787722"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-02-09T19:40:45.057777867"
    },
    "with_guide_tree_versions": {
        "content": [
            [
                "versions.yml:md5,b825152229a974c6cfc6d826db883cb4"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-10T13:29:28.254709"
    }
}